File: frontend/src/lib/calculator-budget-config.ts
export type CalculatorBudgetConfig = {
  dev_rate_rub: number;
  audit_base: number;
  training: number;
  om_base: number;
  capex_min: number;
  integration_mult_simple: number;
  integration_mult_standard: number;
  integration_mult_complex: number;
  integration_cost_2_roles: number;
  integration_cost_3_roles: number;
  integration_cost_4plus_roles: number;
  om_units_500: number;
  om_units_2000: number;
  om_steps_3: number;
  om_steps_5: number;
  om_hm_150: number;
  dev_hours_base: number;
  dev_hours_per_step: number;
  dev_hours_units_500: number;
  dev_hours_units_1000: number;
  dev_hours_units_2000: number;
  dev_hours_automation_high: number;
  dev_hours_hm_80: number;
  dev_hours_hm_200: number;
  automation_pct_threshold: number;
  capex_round_step: number;
  om_round_step: number;
  tier_simple_max: number;
  tier_standard_max: number;
  units_threshold_500: number;
  units_threshold_1000: number;
  units_threshold_2000: number;
  hm_threshold_50: number;
  hm_threshold_80: number;
  hm_threshold_150: number;
  hm_threshold_200: number;
};

export const DEFAULT_CALCULATOR_BUDGET_CONFIG: CalculatorBudgetConfig = {
  dev_rate_rub: 4000,
  audit_base: 100_000,
  training: 40_000,
  om_base: 8_000,
  capex_min: 150_000,
  integration_mult_simple: 0.85,
  integration_mult_standard: 1,
  integration_mult_complex: 1.35,
  integration_cost_2_roles: 40_000,
  integration_cost_3_roles: 80_000,
  integration_cost_4plus_roles: 150_000,
  om_units_500: 4_000,
  om_units_2000: 7_000,
  om_steps_3: 3_000,
  om_steps_5: 5_000,
  om_hm_150: 3_000,
  dev_hours_base: 20,
  dev_hours_per_step: 6,
  dev_hours_units_500: 15,
  dev_hours_units_1000: 30,
  dev_hours_units_2000: 40,
  dev_hours_automation_high: 15,
  dev_hours_hm_80: 10,
  dev_hours_hm_200: 20,
  automation_pct_threshold: 70,
  capex_round_step: 50_000,
  om_round_step: 1_000,
  tier_simple_max: 3,
  tier_standard_max: 6,
  units_threshold_500: 500,
  units_threshold_1000: 1000,
  units_threshold_2000: 2000,
  hm_threshold_50: 50,
  hm_threshold_80: 80,
  hm_threshold_150: 150,
  hm_threshold_200: 200,
};

export type CalculatorBudgetConfigRecord = {
  config: CalculatorBudgetConfig;
  updated_at?: string;
};

export type BudgetConfigFieldKey = keyof CalculatorBudgetConfig;

export const BUDGET_CONFIG_SECTIONS: {
  sectionKey: string;
  fields: BudgetConfigFieldKey[];
}[] = [
  {
    sectionKey: "globalRates",
    fields: ["dev_rate_rub", "audit_base", "training", "om_base", "capex_min"],
  },
  {
    sectionKey: "integration",
    fields: [
      "integration_mult_simple",
      "integration_mult_standard",
      "integration_mult_complex",
      "integration_cost_2_roles",
      "integration_cost_3_roles",
      "integration_cost_4plus_roles",
    ],
  },
  {
    sectionKey: "omAddons",
    fields: ["om_units_500", "om_units_2000", "om_steps_3", "om_steps_5", "om_hm_150"],
  },
  {
    sectionKey: "devHours",
    fields: [
      "dev_hours_base",
      "dev_hours_per_step",
      "dev_hours_units_500",
      "dev_hours_units_1000",
      "dev_hours_units_2000",
      "dev_hours_automation_high",
      "dev_hours_hm_80",
      "dev_hours_hm_200",
      "automation_pct_threshold",
    ],
  },
  {
    sectionKey: "thresholds",
    fields: [
      "tier_simple_max",
      "tier_standard_max",
      "units_threshold_500",
      "units_threshold_1000",
      "units_threshold_2000",
      "hm_threshold_50",
      "hm_threshold_80",
      "hm_threshold_150",
      "hm_threshold_200",
    ],
  },
  {
    sectionKey: "rounding",
    fields: ["capex_round_step", "om_round_step"],
  },
];

export function isIntegerBudgetField(key: BudgetConfigFieldKey): boolean {
  return key.startsWith("tier_");
}

File: backend/internal/model/calculator_budget_config.go
package model

import "time"

type CalculatorBudgetConfig struct {
	DevRateRub              float64 `json:"dev_rate_rub"`
	AuditBase               float64 `json:"audit_base"`
	Training                float64 `json:"training"`
	OmBase                  float64 `json:"om_base"`
	CapexMin                float64 `json:"capex_min"`
	IntegrationMultSimple   float64 `json:"integration_mult_simple"`
	IntegrationMultStandard float64 `json:"integration_mult_standard"`
	IntegrationMultComplex  float64 `json:"integration_mult_complex"`
	IntegrationCost2Roles   float64 `json:"integration_cost_2_roles"`
	IntegrationCost3Roles   float64 `json:"integration_cost_3_roles"`
	IntegrationCost4Plus    float64 `json:"integration_cost_4plus_roles"`
	OmUnits500              float64 `json:"om_units_500"`
	OmUnits2000             float64 `json:"om_units_2000"`
	OmSteps3                float64 `json:"om_steps_3"`
	OmSteps5                float64 `json:"om_steps_5"`
	OmHm150                 float64 `json:"om_hm_150"`
	DevHoursBase            float64 `json:"dev_hours_base"`
	DevHoursPerStep         float64 `json:"dev_hours_per_step"`
	DevHoursUnits500        float64 `json:"dev_hours_units_500"`
	DevHoursUnits1000       float64 `json:"dev_hours_units_1000"`
	DevHoursUnits2000       float64 `json:"dev_hours_units_2000"`
	DevHoursAutomationHigh  float64 `json:"dev_hours_automation_high"`
	DevHoursHm80            float64 `json:"dev_hours_hm_80"`
	DevHoursHm200           float64 `json:"dev_hours_hm_200"`
	AutomationPctThreshold  float64 `json:"automation_pct_threshold"`
	CapexRoundStep          float64 `json:"capex_round_step"`
	OmRoundStep             float64 `json:"om_round_step"`
	TierSimpleMax           int     `json:"tier_simple_max"`
	TierStandardMax         int     `json:"tier_standard_max"`
	UnitsThreshold500       float64 `json:"units_threshold_500"`
	UnitsThreshold1000      float64 `json:"units_threshold_1000"`
	UnitsThreshold2000      float64 `json:"units_threshold_2000"`
	HmThreshold50           float64 `json:"hm_threshold_50"`
	HmThreshold80           float64 `json:"hm_threshold_80"`
	HmThreshold150          float64 `json:"hm_threshold_150"`
	HmThreshold200          float64 `json:"hm_threshold_200"`
}

type CalculatorBudgetConfigRecord struct {
	Config    CalculatorBudgetConfig `json:"config"`
	UpdatedAt time.Time              `json:"updated_at"`
}

func DefaultCalculatorBudgetConfig() CalculatorBudgetConfig {
	return CalculatorBudgetConfig{
		DevRateRub:              4000,
		AuditBase:               100_000,
		Training:                40_000,
		OmBase:                  8_000,
		CapexMin:                150_000,
		IntegrationMultSimple:   0.85,
		IntegrationMultStandard: 1,
		IntegrationMultComplex:  1.35,
		IntegrationCost2Roles:   40_000,
		IntegrationCost3Roles:   80_000,
		IntegrationCost4Plus:    150_000,
		OmUnits500:              4_000,
		OmUnits2000:             7_000,
		OmSteps3:                3_000,
		OmSteps5:                5_000,
		OmHm150:                 3_000,
		DevHoursBase:            20,
		DevHoursPerStep:         6,
		DevHoursUnits500:        15,
		DevHoursUnits1000:       30,
		DevHoursUnits2000:       40,
		DevHoursAutomationHigh:  15,
		DevHoursHm80:            10,
		DevHoursHm200:           20,
		AutomationPctThreshold:  70,
		CapexRoundStep:          50_000,
		OmRoundStep:             1_000,
		TierSimpleMax:           3,
		TierStandardMax:         6,
		UnitsThreshold500:       500,
		UnitsThreshold1000:      1000,
		UnitsThreshold2000:      2000,
		HmThreshold50:           50,
		HmThreshold80:           80,
		HmThreshold150:          150,
		HmThreshold200:          200,
	}
}
